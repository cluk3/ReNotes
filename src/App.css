/* 
.toolbar button:hover > svg {
fill: #dcdcdc;
} */