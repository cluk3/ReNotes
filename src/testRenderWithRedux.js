import React from 'react';
import { createStore } from 'redux';
import { Provider, connect } from 'react-redux';
import { render, fireEvent, cleanup } from '@testing-library/react';
import { initialState as notesInitialState } from './Notes/modules/notes';
import { initialState as foldersInitialState } from './Folders/modules/folders';
import { initialState as focusedElementInitialState } from './containers/Focusable/stateManager';

const reducer = state => state;
const defaultInitialState = {
  folders: foldersInitialState,
  notes: notesInitialState,
  focusedElement: focusedElementInitialState
};

export function renderWithRedux(
  ui,
  {
    initialState = defaultInitialState,
    store = createStore(reducer, initialState)
  } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store
  };
}
