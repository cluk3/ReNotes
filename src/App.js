import React, { Component } from 'react';
import './App.css';
import FoldersList from './Folders/FoldersList';
import NotesList from './Notes/NotesList';
import NoteEditor from './Editor/NoteEditor';
import styled from 'styled-components';
import Toolbar from './Toolbar/Toolbar';
import NotesToolbar from './Notes/NotesToolbar';
import Focusable from './containers/Focusable';
import { ENTITIES } from './constants';
import { DndProvider } from 'react-dnd';
import HTML5Backend from 'react-dnd-html5-backend';

const commonColumnStyles = `
  display: block;
  position: relative;
  overflow: hidden;
`;

const FoldersColumn = styled.div`
  ${commonColumnStyles};
  background-color: #f3f4f5;
`;

const NotesColumn = styled.div`
  ${commonColumnStyles};
  background-color: #f9f9f7;
`;

const EditorColumn = styled.div`
  ${commonColumnStyles};
  background-color: #f9f9f7;
`;

const ColumnsContainer = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  position: absolute;
  top: 0;
  bottom: 0;
  width: 100%;
`;

class App extends Component {
  render() {
    return (
      <div className="App" style={{ height: '100vh' }}>
        <DndProvider backend={HTML5Backend}>
          <ColumnsContainer>
            <FoldersColumn key={0}>
              <Toolbar />
              <Focusable elementType={ENTITIES.FOLDERS}>
                <FoldersList />
              </Focusable>
            </FoldersColumn>
            <NotesColumn key={1}>
              <NotesToolbar />
              <Focusable elementType={ENTITIES.NOTES}>
                <NotesList />
              </Focusable>
            </NotesColumn>
            <EditorColumn key={2}>
              <Toolbar />
              <Focusable elementType={ENTITIES.EDITOR}>
                <NoteEditor />
              </Focusable>
            </EditorColumn>
          </ColumnsContainer>
        </DndProvider>
      </div>
    );
  }
}

export default App;
