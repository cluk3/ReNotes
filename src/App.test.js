import React from 'react';
import { renderWithRedux } from './testRenderWithRedux'
import App from './App';
import 'jest-styled-components';

it('renders the App', () => {
  const { container } = renderWithRedux(<App />);
  expect(container.querySelector('.App')).toBeInTheDocument();
});

test('it works', () => {
    const { container } = renderWithRedux(<App />);
    expect(container.firstChild).toMatchSnapshot()
})

it('renders the folders column', () => {
    const { store, getByText } = renderWithRedux(<App />);
    console.log(store)
    expect(getByText(store.getState().folders.byId["0"].name)).toBeInTheDocument();
});