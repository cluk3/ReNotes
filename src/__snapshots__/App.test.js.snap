// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  position: absolute;
  top: 0;
  bottom: 0;
  width: 100%;
}

.c1 {
  display: block;
  position: relative;
  overflow: hidden;
  background-color: #f3f4f5;
}

.c2 {
  height: 36px;
  background-color: #dcdcdc;
  color: white;
  border-bottom: #a9a9a9 solid 1px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  height: 100%;
}

.c3:focus {
  outline: none;
}

.c4 {
  height: 100%;
  overflow-y: auto;
  border-right: #dedede solid 1px;
}

.c5 {
  margin: 0;
  padding: 0;
}

.c6 {
  list-style: none;
  padding: 0.4em 0 0.4em 1em;
  background-color: #3b90fe;
}

.c7 {
  list-style: none;
  padding: 0.4em 0 0.4em 1em;
  background-color: #f3f4f5;
}

.c8 {
  position: absolute;
  bottom: 0;
  border: none;
  font-size: 16px;
  background-color: #f3f4f5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.5em 0.5em;
}

.c9 {
  fill: #888889;
}

.c10 {
  padding-left: 0.3em;
}

.c11 {
  display: block;
  position: relative;
  overflow: hidden;
  background-color: #f9f9f7;
}

.c12 {
  margin-right: 0.5em;
  border-radius: 0.5em;
  padding: 0.2em 1.2em;
  min-height: 2.4em;
  outline: none;
}

.c12:active {
  background-color: #dcdcdc;
  outline: none;
}

.c12:hover > svg {
  fill: #dcdcdc;
}

.c13 {
  max-height: 100%;
  overflow-y: auto;
}

.c14 {
  background-color: #dfdfdd;
  padding-left: 1em;
}

.c20 {
  background-color: #f9f9f7;
  padding-left: 1em;
}

.c15 {
  font-size: 14px;
  padding: 1em 1em 1em 0;
  border-bottom: #dedede solid 1px;
}

.c16 {
  font-size: 15px;
  margin: 0;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  width: 100%;
  line-height: 1.4;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
}

.c18 {
  padding-right: 1em;
  white-space: nowrap;
}

.c19 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c21 {
  display: block;
  position: relative;
  overflow: hidden;
  background-color: #f9f9f7;
}

.c22 {
  height: 100%;
  max-height: 100%;
  overflow: auto;
  padding: 0.5em 1em 1em 1em;
  border-left: #dedede solid 1px;
}

.c23 {
  margin-bottom: 1em;
  display: block;
  text-align: center;
  color: #9ca09f;
  font-size: 14px;
  font-weight: 300;
}

<div
  class="App"
  style="height: 100vh;"
>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
      <div
        class="c3"
        tabindex="-1"
      >
        <div
          class="c4"
        >
          <ul
            class="c5"
          >
            <li
              class="c6"
            >
              Test Folder
            </li>
            <li
              class="c7"
            >
              Test Folder 2
            </li>
          </ul>
          <button
            class="c8"
          >
            <svg
              class="c9"
              height="20px"
              viewBox="0 0 30 30"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15,3C8.373,3,3,8.373,3,15c0,6.627,5.373,12,12,12s12-5.373,12-12C27,8.373,21.627,3,15,3z M21,16h-5v5 c0,0.553-0.448,1-1,1s-1-0.447-1-1v-5H9c-0.552,0-1-0.447-1-1s0.448-1,1-1h5V9c0-0.553,0.448-1,1-1s1,0.447,1,1v5h5 c0.552,0,1,0.447,1,1S21.552,16,21,16z"
              />
            </svg>
            <span
              class="c10"
            >
              New Folder
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="c11"
    >
      <div
        class="c2"
      >
        <button
          class="c12"
        >
          <img
            alt="delete"
            height="18px"
            src="notes-delete-icon.png"
          />
        </button>
        <button
          class="c12"
        >
          <svg
            class="c9"
            height="20px"
            viewBox="0 0 30 30"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15,3C8.373,3,3,8.373,3,15c0,6.627,5.373,12,12,12s12-5.373,12-12C27,8.373,21.627,3,15,3z M21,16h-5v5 c0,0.553-0.448,1-1,1s-1-0.447-1-1v-5H9c-0.552,0-1-0.447-1-1s0.448-1,1-1h5V9c0-0.553,0.448-1,1-1s1,0.447,1,1v5h5 c0.552,0,1,0.447,1,1S21.552,16,21,16z"
            />
          </svg>
        </button>
      </div>
      <div
        class="c3"
        tabindex="-1"
      >
        <div
          class="c13"
        >
          <div
            class="c14"
          >
            <div
              class="c15"
            >
              <h2
                class="c16"
              >
                lorem ipsum foo bar sit dolor amen lorepsum foo bar sit dolor amen
              </h2>
              <div
                class="c17"
              >
                <span
                  class="c18"
                >
                  Yesterday
                </span>
                <span
                  class="c19"
                >
                  No additional text
                </span>
              </div>
            </div>
          </div>
          <div
            class="c20"
          >
            <div
              class="c15"
            >
              <h2
                class="c16"
              >
                lorem ipsum foo bar sit
              </h2>
              <div
                class="c17"
              >
                <span
                  class="c18"
                >
                  Friday
                </span>
                <span
                  class="c19"
                >
                  No additional text
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c21"
    >
      <div
        class="c2"
      />
      <div
        class="c3"
        tabindex="-1"
      >
        <div
          class="c22"
        >
          <span
            class="c23"
          >
            23 June 2019 at 5:07 PM
          </span>
          <div
            class="DraftEditor-root"
          >
            <div
              class="DraftEditor-editorContainer"
            >
              <div
                class="notranslate public-DraftEditor-content"
                contenteditable="true"
                role="textbox"
                spellcheck="false"
                style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
              >
                <div
                  data-contents="true"
                >
                  <div
                    class=""
                    data-block="true"
                    data-editor="123"
                    data-offset-key="123-0-0"
                  >
                    <div
                      class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                      data-offset-key="123-0-0"
                    >
                      <span
                        data-offset-key="123-0-0"
                      >
                        <span
                          data-text="true"
                        >
                          lorem ipsum foo bar sit dolor amen lorepsum foo bar sit dolor amen
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
